Makefile.toml

```diff
@@ -83,3 +120,66 @@ proto-gen \
 install_crate = { crate_name = "git-cliff", binary = "git-cliff", test_arg = ["-V"], min_version = "1.4.0" }
 command = "git-cliff"
 args = ["-o", "CHANGELOG.md", "-t", "${@}"]
+
+### CI TASKS
+
+[tasks.ci-workspace]
+run_task = { name = [
+    "check-lockfile-patches",
+    "fmt",
+    "clippy",
+    "check-lockfile",
+] }
+
+[tasks.ci-standalone]
+# Provide paths to crates to test
+# Example: cargo make ci-standalone services/shuttle-axum
+script_runner = "@duckscript"
+script = '''
+alias run exec --fail-on-error
+
+for path in ${@}
+
+
+if not ${path}
+    echo No crate path provided
+    exit 1
+end
+manifest = set "${path}/Cargo.toml"
+if not is_path_exists ${manifest}
+    echo manifest not found: ${manifest}
+    exit 1
+end
+
+feature_flags = array "--all-features"
+###### Crates that have mutually exclusive features define
+###### their own set of feature flags to test:
+if eq ${path} "resources/shared-db"
+    feature_flags = array "-F mongodb" "-F postgres" "-F postgres,sqlx" "-F postgres,sqlx-native-tls"
+elseif eq ${path} "services/shuttle-axum"
+    feature_flags = array "-F axum" "-F axum-0-6"
+elseif eq ${path} "services/shuttle-serenity"
+    feature_flags = array "-F serenity,rustls_backend" "-F serenity,native_tls_backend" "-F serenity-0-11-rustls_backend" "-F serenity-0-11-native_tls_backend"
+end
+
+echo
+println -s bold "### Manifest:"
+println -s bold "    ${manifest}"
+echo
+
+run cargo fmt --all --check --manifest-path ${manifest}
+
+for f in ${feature_flags}
```

Comment:
Didn't the old, CI tests run in parallel? Ie isn't this slowing down the CI?

Reply:
Yes, however, these crates are small and generally complete before everything else. Since 90%+ of the dep tree is being compiled for one feature flag set, it is wasteful to not re-use the build output for the other feature flag sets.  The previous approach would get out of hand quickly if the matrix of flag sets to support grows.
This approach can also be taken to test more than one crate per job, but I think we are at a good middle ground here.

Reply:
I see. Sounds good!